shared/src/commonMain/kotlin/org/hyperskill/app/study_plan/widget/presentation/StudyPlanWidgetActionDispatcher.kt

```diff
@@ -15,10 +16,13 @@ class StudyPlanWidgetActionDispatcher(
 ) : CoroutineActionDispatcher<Action, Message>(config.createConfig()) {
     override suspend fun doSuspendableAction(action: Action) {
         when (action) {
-            InternalAction.FetchStudyPlan -> {
+            is InternalAction.FetchStudyPlan -> {
+                if (action.delay != null) {
+                    delay(action.delay.inWholeMilliseconds)
```

Comment:
I need to check will `delay` block the main thread on iOS

Reply:
In case of Subssion polling delay is used.
https://github.com/hyperskill/mobile-app/blob/fc7942a058e447c1d1728e8098ac09b61d0f1a17/shared/src/commonMain/kotlin/org/hyperskill/app/step_quiz/domain/interactor/StepQuizInteractor.kt#L68-L81

Reply:
According to [the documentation](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/delay.html), `delay` suspend the coroutine without blocking the thread.

The source code shows it:
[Delay](https://github.com/Kotlin/kotlinx.coroutines/blob/ca14606d1f60780663dbcf54b58d2bb2b5e10f7d/kotlinx-coroutines-core/common/src/Delay.kt#L125-133) function internally calls [scheduleResumeAfterDelay](https://github.com/Kotlin/kotlinx.coroutines/blob/ca14606d1f60780663dbcf54b58d2bb2b5e10f7d/kotlinx-coroutines-core/common/src/Delay.kt#L49)
[DarwinMainDispatcher](https://github.com/Kotlin/kotlinx.coroutines/blob/ca14606d1f60780663dbcf54b58d2bb2b5e10f7d/kotlinx-coroutines-core/nativeDarwin/src/Dispatchers.kt#L47-L65) implement `Delay` interface.

Reply:
Checked, not blocks